tests/lib/devicetree/api/src/main.c

```diff
@@ -1387,6 +1424,13 @@ static void test_devices(void)
 #define DT_DRV_COMPAT xxxx
 #define BUILD_BUG_ON_EXPANSION (there is a bug in devicetree.h)
 	DT_INST_FOREACH_STATUS_OKAY(BUILD_BUG_ON_EXPANSION)
+#undef BUILD_BUG_ON_EXPANSION
+
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT xxxx
```

Comment:
Why undefine it and define it to the same thing again?

Reply:
I wanted to keep the test for each feature clean and separate.

Reply:
They are so close to each other that I think it's probably more confusing than just leaving it as-is, but okay, I accept that.